### ISSUE TITLE
Add Dockerfile

### ISSUE THREAD

**Issue by codetheweb** [Kiro47/MTU-Transfer-Course-Gatherer#31]
Waiting on #29.

**Comment by Blu3spirits** [comment]
Taking this since I've got a docker file already that I use to deploy to my k8s cluster. Still need to wait on #29 PR though

**Comment by Blu3spirits** [comment]
Copy of the Dockerfile I sent to code,
```
FROM python:buster

RUN apt-get update

RUN apt-get -y install git rsync

RUN git clone https://github.com/Kiro47/MTU-Transfer-Course-Gatherer.git /app/
WORKDIR /app/


# Install Python dependencies
RUN pip3 install --upgrade pip -r requirements.txt
# Install Python deployment packages (Issue #29)
RUN pip3 install gunicorn whitenoise
RUN python3 ./manage.py makemigrations
RUN python3 ./manage.py migrate
# This takes forever, if you have a writer.csv just copy it in
RUN python3 ./banwebScrape.py
RUN python3 ./manage.py custom_db_sync /app/writer.csv

# Set django settings to PROD
ENV DJANGO_SETTINGS_MODULE="course_gather.settings.prod_settings"
ENV REACT_APP_ENDPOINT="https://localhost:8000/"
WORKDIR /app/webapp/

# Npm production build
RUN npm run build
RUN mkdir -p /app/static/

# Npm run build lays the structure out like build/<files> build/static
RUN rsync -av --progress --exclude=static/ build/ /app/static/
RUN rsync -av --progress build/static/* /app/static/

WORKDIR /app/
EXPOSE 8000

# Use gunicorn to actually run the wsgi server
CMD ["gunicorn", "--workers", "4", "--bind", ":8000", "--log-level", "INFO", "course_gather.wsgi:application"]
```

**Comment by Blu3spirits** [comment]
```
FROM python:buster

RUN apt-get update

RUN apt-get -y install curl git rsync \
    && curl -sL https://deb.nodesource.com/setup_14.x | bash \
    && apt-get install nodejs \
    && curl -o- -L https://yarnpkg.com/install.sh | bash


#RUN git clone https://github.com/codetheweb/MTU-Transfer-Course-Gatherer.git /app/
COPY . /app
WORKDIR /app/

#RUN git checkout feature/ui-touchup


# Install Python dependencies
# COPY ./backend/requirements.txt /app/backend/
RUN pip3 install --upgrade pip -r requirements.txt
RUN pip3 install gunicorn whitenoise
RUN python3 ./manage.py makemigrations
RUN python3 ./manage.py migrate
# RUN python3 ./banwebScrape.py
RUN python3 ./manage.py custom_db_sync /app/writer.csv
ENV DJANGO_SETTINGS_MODULE="course_gather.settings.prod_settings"
ENV REACT_APP_ENDPOINT="$ENDPOINT"
WORKDIR /app/webapp/
RUN npm install --only=prod
RUN npm run build
RUN mkdir -p /app/static/
RUN rsync -av --progress --exclude=static/ build/ /app/static/
RUN rsync -av --progress build/static/* /app/static/
# RUN cp -v build/* /app/static/
# RUN cp -vr build/static/* /app/static/
WORKDIR /app/
EXPOSE 8000
```

Code's dockerfile from his deployment

Notes:
- Proxy isn't needed! :D 
- add whitenoise + gunicorn to requirements
- ENV django settings maybe to arg?
- ARG ENDPOINT variable
- add gunicorn config file

**Comment by codetheweb** [comment]
Also, might want to use multi-stage builds to get rid of Node for a smaller final image.
Probably doesn't matter as much since the config vars are embedded into the build and thus we can't publish it to Docker Hub.

**Comment by codetheweb** [comment]
Resolved by #33.
